sensor/kubernetes/complianceoperator/handler_impl.go

```diff
@@ -0,0 +1,293 @@
+	}
+}
+
+func (m *handlerImpl) ResponsesC() <-chan *central.MsgFromSensor {
+	return m.response
+}
+
+func (m *handlerImpl) run() {
+	for !m.stopSignal.IsDone() {
+		select {
+		case req := <-m.request:
+			var requestProcessed bool
+			switch r := req.GetRequest().(type) {
+			case *central.ComplianceRequest_EnableCompliance:
+				requestProcessed = m.enableCompliance(r.EnableCompliance)
+			case *central.ComplianceRequest_DisableCompliance:
+				requestProcessed = m.disableCompliance(r.DisableCompliance)
+			case *central.ComplianceRequest_ApplyScanConfig:
+				requestProcessed = m.processApplyScanCfgRequest(r.ApplyScanConfig)
+			case *central.ComplianceRequest_DeleteScanConfig:
+				requestProcessed = m.processDeleteScanCfgRequest(r.DeleteScanConfig)
+			}
+
+			if !requestProcessed {
+				log.Errorf("Could not send response for compliance request: %s", protoutils.NewWrapper(req))
+			}
+		case <-m.stopSignal.Done():
+			return
+		}
+	}
+}
+
+func (m *handlerImpl) enableCompliance(request *central.EnableComplianceRequest) bool {
+	m.disabled.Reset()
+	// TODO: [ROX-18096] Start collecting compliance profiles & rules
+	return m.composeAndSendEnableComplianceResponse(request.GetId(), nil)
+}
+
+func (m *handlerImpl) disableCompliance(request *central.DisableComplianceRequest) bool {
+	// Disabling compliance should not disable compliance operator status monitoring. Users must be informed about
+	// the compliance operator health.
+
+	// TODO: Pause all scans. Currently, the only way a compliance scan can be disabled is by deleting the ScanSettingBinding.
+	// TODO: Drop all custom resource listener events.
+	m.disabled.Signal()
+	return m.composeAndSendDisableComplianceResponse(request.GetId(), nil)
+}
+
+func (m *handlerImpl) processApplyScanCfgRequest(request *central.ApplyComplianceScanConfigRequest) bool {
+	select {
+	case <-m.disabled.Done():
+		err := errors.Errorf("Compliance is disabled. Cannot process request: %s", protoutils.NewWrapper(request))
+		return m.composeAndSendApplyScanConfigResponse(request.GetId(), err)
+	case <-m.stopSignal.Done():
+		return true
+	default:
+		if request.GetScanRequest() == nil {
+			return m.composeAndSendApplyScanConfigResponse(request.GetId(), errors.New("Compliance scan request is empty"))
+		}
+		if req := request.GetOneTimeScan(); req != nil {
+			return m.processOneTimeScanRequest(request.GetId(), req)
+		}
+		return m.composeAndSendApplyScanConfigResponse(request.GetId(), errors.New("Cannot handle compliance scan request"))
+	}
+}
+
+func (m *handlerImpl) processOneTimeScanRequest(requestID string, request *central.ApplyComplianceScanConfigRequest_OneTimeScan) bool {
+	if err := validateApplyOneTimeScanConfigRequest(request); err != nil {
+		return m.composeAndSendApplyScanConfigResponse(requestID, errors.Wrap(err, "validating compliance scan request"))
+	}
+	// TODO: Check if default ACS scan setting CR exists. If it doesn't exist, create one.
```

Comment:
That could be interesting.  If the only way to disable a scan is to delete the scan settings and we allow the user to turn off the default scan how will sensor know to create it or not?  Perhaps we should bundle and store the default scan setting schedule in central.  That way as far as sensor is concerned it is just another scan setting.